duffel_api/utils.py

```diff
@@ -1,12 +1,13 @@
 """Assorted auxiliary functions"""
-from datetime import datetime, date
+from datetime import date, datetime
+from typing import Any, Union
 
 
-def maybe_parse_date_entries(key, value):
-    """Parse datetime entries, depending on the value of `key`"""
-    if value is None:
+def maybe_parse_date_entries(key: str, value: Any) -> Union[str, datetime, date]:
```

Comment:
The `value` is always string, is it not?

Reply:
I believe it can be `None` too.

Reply:
Yes, I guess the line I'd previously added above illustrates that pretty well! 🤣

Reply:
Hahah, I was doubting whether that was true though!